### ISSUE TITLE
Intellisense broken on Unix-like OSes

### ISSUE THREAD

**Issue by maxgerhardt** [platformio/builder-framework-arduino-core-mbed#4]
Using

```ini
[env:pico]
platform = https://github.com/platformio/platform-raspberrypi.git
board = pico
framework = arduino
```
(this means ArduinoCore-mbed is used and the builder script in this repo), on Linux and Mac machines with the latest VSCode 

```
Version: 1.68.0
Commit: 4af164ea3a06f701fe3e89a2bcbb421d2026b68f
Date: 2022-06-08T11:49:57.055Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Linux x64 5.4.0-58-generic
```

and Microsoft C/C++ extension version 1.10.5 and PlatformIO extension version 2.4.3, the Intellisense is broken, while building works.

![grafik](https://user-images.githubusercontent.com/26485477/173247540-121201ab-7827-4107-9b21-5f59727c50a7.png)

^-- Specifically tested on Ubuntu 20.04.4 LTS.

As reported [in the forum](https://community.platformio.org/t/intellisense-include-errors-with-arduino-mbed-framework-and-raspberry-pi-pico/28294).

Note that IntelliSense works fine on Windows with the same VSCode version. I am not sure why. There seems to be a fundamental problem here in the Intellisense behavior.

**Comment by maxgerhardt** [comment]
Update: This seems to be interconnected with the Microsoft C/C++ extension version:
* 1.10.5 (latest): fails
* 1.9.8 (previously released version): good
* 1.9.7: good
* 1.8.4: good

**Comment by ivankravets** [comment]
We can't fix this issue on our side. There is a CLI MAX_LEN limitation on Windows.

Let me ping the Microsoft team to fix https://github.com/microsoft/vscode-cpptools/issues/9434 ASAP. Thanks for reporting this issue to the `vscode-cpptools`.

**Comment by maxgerhardt** [comment]
Technically, the fix PR would still use `-iprefix` for building, but when the target is `_idedata`, it would read that file and expose the paths in `env[CPPPATH]` -- so building would not break, but individual paths would show up in Intellisense. But indeed, Microsoft bugs should be fixed **instead**.

**Comment by valeros** [comment]
Hi @maxgerhardt , thanks for reporting. Should be fixed in the latest VSCode C/C++ extension [v1.10.6](https://github.com/microsoft/vscode-cpptools/releases/tag/v1.10.6). As for the `_idedata` target, it seems like an internal target used by PlatformIO core, so it might not be the best idea to use it in build scripts.
